core/src/main/java/org/apache/iceberg/ManifestFileParser.java

```diff
@@ -0,0 +1,277 @@
+    if (jsonNode.has(DELETED_FILES_COUNT)) {
+      deletedFilesCount = JsonUtil.getInt(DELETED_FILES_COUNT, jsonNode);
+    }
+
+    Long addedRowsCount = null;
+    if (jsonNode.has(ADDED_ROWS_COUNT)) {
+      addedRowsCount = JsonUtil.getLong(ADDED_ROWS_COUNT, jsonNode);
+    }
+
+    Long existingRowsCount = null;
+    if (jsonNode.has(EXISTING_ROWS_COUNT)) {
+      existingRowsCount = JsonUtil.getLong(EXISTING_ROWS_COUNT, jsonNode);
+    }
+
+    Long deletedRowsCount = null;
+    if (jsonNode.has(DELETED_ROWS_COUNT)) {
+      deletedRowsCount = JsonUtil.getLong(DELETED_ROWS_COUNT, jsonNode);
+    }
+
+    List<ManifestFile.PartitionFieldSummary> partitionFieldSummaries = null;
+    if (jsonNode.has(PARTITION_FIELD_SUMMARY)) {
+      JsonNode summaryArray = jsonNode.get(PARTITION_FIELD_SUMMARY);
+      Preconditions.checkArgument(
+          summaryArray.isArray(),
+          "Invalid JSON node for partition field summaries: non-array (%s)",
+          summaryArray);
+
+      ImmutableList.Builder<ManifestFile.PartitionFieldSummary> builder = ImmutableList.builder();
+      for (JsonNode summaryNode : summaryArray) {
+        ManifestFile.PartitionFieldSummary summary =
+            PartitionFieldSummaryParser.fromJson(summaryNode);
+        builder.add(summary);
+      }
+
+      partitionFieldSummaries = builder.build();
+    }
+
+    ByteBuffer keyMetadata = null;
+    if (jsonNode.has(KEY_METADATA)) {
+      String hexStr = JsonUtil.getString(KEY_METADATA, jsonNode);
+      keyMetadata = ByteBuffer.wrap(BaseEncoding.base16().decode(hexStr));
+    }
+
+    return new GenericManifestFile(
+        path,
+        length,
+        specId,
+        manifestContent,
+        sequenceNumber,
+        minSequenceNumber,
+        addedSnapshotId,
+        partitionFieldSummaries,
+        keyMetadata,
+        addedFilesCount,
+        addedRowsCount,
+        existingFilesCount,
+        existingRowsCount,
+        deletedFilesCount,
+        deletedRowsCount);
+  }
+
+  private static class PartitionFieldSummaryParser {
+    private static final String CONTAINS_NULL = "contains-null";
+    private static final String CONTAINS_NAN = "contains-nan";
+    private static final String LOWER_BOUND = "lower-bound";
+    private static final String UPPER_BOUND = "upper-bound";
+
+    private PartitionFieldSummaryParser() {}
+
+    static void toJson(ManifestFile.PartitionFieldSummary summary, JsonGenerator generator)
+        throws IOException {
+      Preconditions.checkArgument(summary != null, "Invalid partition field summary: null");
+      Preconditions.checkArgument(generator != null, "Invalid JSON generator: null");
+
+      generator.writeStartObject();
+
+      generator.writeBooleanField(CONTAINS_NULL, summary.containsNull());
+
+      if (summary.containsNaN() != null) {
+        generator.writeBooleanField(CONTAINS_NAN, summary.containsNaN());
+      }
+
+      if (summary.lowerBound() != null) {
+        generator.writeStringField(
```

Comment:
should this be doing the same thing as in https://github.com/apache/iceberg/blob/ccaeb2f4d5be44b4ab177b8fb790194eff8070a5/core/src/main/java/org/apache/iceberg/ContentFileParser.java#L97?